src/libs/rover/rover.cpp

```diff
@@ -226,6 +226,7 @@ Rover::~Rover()
 void
 Rover::update_settings(Node &params)
 {
+  // TODO: Change this to use has_child everywhere instead
   if (params.has_path("camera/look_at"))
```

Comment:
if you're curious about examining performance of checking `has_child` versus `has_path`, that's fine and probably useful. But just calling a straight `set` is going to be faster than both of those approaches. We don't need to ask what is in the params, we can just add them directly to `m_settings`.

Reply:
I'm just suggesting we do this in place of the entire function: https://github.com/Alpine-DAV/ascent/pull/1541#discussion_r2117004224